src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,14 +242,47 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
+            QVariantMap dmap;
+
             if(data)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                /* determine the profile to use */
+                dmap["db"] = QVariant ((qulonglong)data);
+
+                IAppSetup *setup = IAppSetup::getPlatformInstance();
+
+                /* check possible profiles.xml locations and keep the first active one. */
+                dmap["prefixpath"] = "";
+                QStringList profilespaths = {
+                    dir.filePath(prefix+"-profiles.xml"),
+                    dir.filePath("profiles.xml"),
+                    setup->routinoPath("profiles.xml").toUtf8()
+                };
+                QStringListIterator ppit(profilespaths);
+                while( ppit.hasNext() )
+                {
+                    QFileInfo pinfo = QFileInfo(ppit.next());
+                    if( pinfo.isReadable())
+                    {
+                        dmap["prefixpath"] = pinfo.filePath();
+                        break;
+                    }
+                }
+                if( dmap["prefixpath"].toString().length() == 0)
+                {
+                    QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
+                }
+
+                qDebug() << "Profile ... Using \n" << dmap["prefixpath"].toString();
+
+                comboDatabase->addItem(prefix.replace("_", " "), dmap);
             }
             else
```

Comment:
As you added quite a bunch of code the the true path the reason for the else path is harder to comprehend. Please move it before `if(data)`

```
if(data == nullptr)
{
    QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
    continue;
}
QVariantMap dmap;
dmap["db"] = QVariant ((qulonglong)data);
...
```

Reply:
Done - much cleaner.  Negates a comment above.